src/jetson/main.py

```diff
@@ -227,54 +225,62 @@ def classifyFrame(self,
+        '''
+        self.encryptor = AESEncryptor()
+        self.key = self.encryptor.key
+ 
 
-def encryptFace(coordinates: List[Tuple[int]],
-                img: np.ndarray):
-    '''
-    This function Encrypts faces
-    Args:
-        coordinates - Face coordinates returned by face detector
-        img - A 3D numpy array containing image to be encrypted
-
-    Return:
-        encryptedImg - Image with face coordinates encrypted
-    '''
-
-    encryptor = Encryptor()
-    encryptedImg, _ = encryptor.encrypt(coordinates, img)
-
-    return encryptedImg
-
-def encryptFrame(img:np.ndarray,
-                boxes:List[Tuple[np.float64]]):
-    '''
-    This method takes the face coordinates, encrypts the facial region, writes encrypted image to file filesystem
-    Args:
-        img: A 3D numpy array containing image to be encrypted
-        boxes: facial Coordinates
-    '''
-    try:
+    def encryptFace(self, coordinates: List[Tuple[int]],
+                    img: np.ndarray):
+        '''
+        This function Encrypts faces
+        Args:
+            coordinates - Face coordinates returned by face detector
+            img - A 3D numpy array containing image to be encrypted
+    
+        Return:
+            encryptedImg - Image with face coordinates encrypted
+        '''
+    
+        encryptedImg, _ = self.encryptor.encrypt(coordinates, img)
+    
+        return encryptedImg
+    
+    def encryptFrame(self, img:np.ndarray,
+                    boxes:List[Tuple[np.float64]],
+                    main_process):
+        '''
+        This method takes the face coordinates, encrypts the facial region, writes encrypted image to file filesystem
+        Args:
+            img: A 3D numpy array containing image to be encrypted
+            boxes: facial Coordinates
+        '''
         for box in boxes:
             x1, y1, x2, y2 = [int(b) for b in box]
             # draw boxes within the frame
             x1 = max(0, x1)
             y1 = max(0, y1)
             x2 = min(img.shape[1], x2)
             y2 = min(img.shape[0], y2)
-
-            img = encryptFace([(x1, y1, x2, y2)], img)
-
+    
+            img = self.encryptFace([(x1, y1, x2, y2)], img)
+    
         #TODO ftp img to remote
-        #Lets just write img to filesystem for now
-        global fileCount
-        face_file_name = os.path.join(args.output_dir, f'{fileCount}.jpg')
-
-        #TODO: Remove this print statement after db integration
-        print("writing ", face_file_name)
-        fileCount += 1
-        cv2.imwrite(face_file_name, img)
-    except KeyboardInterrupt:
-        pass
+        if args.write_imgs:
+            if not os.path.isdir(args.output_dir):
+                os.mkdir(args.output_dir)
+            global fileCount
+            face_file_name = os.path.join(args.output_dir, f'{fileCount.value}.jpg')
+    
+            #TODO: Remove this print statement after db integration
```

Comment:
can we please address this. either resolve or notify the person in charge of resolving this

Reply:
Could you clarify what needs to be resolved? db integration has not yet been implemented

Reply:
Is db integration being worked on then? Sorry I'm not at the meetings so I am a bit clueless to what is being worked on at the moment. If db integration does not exist can you please provide a comment to how this code behaves without db integration and why its needed. Sometimes when features like this get pushed to the side then the person who wrote this comment might not be around to answer why they wrote it if it becomes ambiguous to why it was written in the first place.

As a good rule of thumb for comments. don't say what the code is doing or what the code needs to do in the future, say why the code is doing what its doing. for example.

```python
# TODO: add db integration here 
```

is a lot more ambiguous than 

```python
# TODO: add db integration here so we can store the images collected for future analysis
```

See how the second comment says what the reason for adding db integration was? it makes it a lot easier to understand and potentially saves a meeting in the future by simply giving more context to this comment.

Reply:
Thanks for the feedback. db integration into main.py is planned after we can get it cleaned up and implement processing changes. We don't intend to replace any of the code by this TODO, just remove the print statement: it only exists now for testing purposes